components/hab/mac/mac-build.sh

```diff
@@ -30,33 +30,34 @@ install_if_missing() {
     local formula="$pkg"
   fi
 
-  echo $(sudo -u ${SUDO_USER} brew list --versions ${pkg})
+  sudo -u "${SUDO_USER}" brew list --versions "${pkg}"
 
-  if [[ $(sudo -u ${SUDO_USER} brew list --versions $pkg | wc -l) -eq 0 ]]; then
+  if [[ $(sudo -u "${SUDO_USER}" brew list --versions "$pkg" | wc -l) -eq 0 ]]; then
     info "Installing missing Homebrew package $formula"
-    sudo -u $SUDO_USER brew install $formula
+    sudo -u "$SUDO_USER" brew install "$formula"
   fi
 }
 
-if (( $EUID != 0 )); then
+if (( EUID != 0 )); then
   info "Please run as root (with \`sudo $0 $*\`)"
   exit 1
 fi
 
 if [[ ! -f /usr/local/bin/hab ]]; then
   info "Habitat CLI missing, attempting to install latest release"
   mkdir -p /usr/local/bin
-  sh $(dirname $0)/../install.sh
+  sh "$(dirname "$0")"/../install.sh
 fi
 
 while true; do
+  # shellcheck disable=2012
```

Comment:
```
In ./components/hab/mac/mac-build.sh line 53:
  if [[ $(ls -1 /hab/cache/keys/core-*.sig.key 2> /dev/null | wc -l) -gt 0 ]]; then
          ^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.
```
See https://github.com/koalaman/shellcheck/wiki/SC2012

This could be safer and simpler as either:
```bash
  if stat /hab/cache/keys/core-*.sig.key &> /dev/null; then
```
or
```
  if [[ $(find /hab/cache/keys/ -name "core-*.sig.key") ]]; then
```
Which is clearer?

Reply:
I like the `find` formulation, personally

Reply:
I also prefer the `find` version